frontend/src/components/ViewActivities/index.js

```diff
@@ -8,12 +15,75 @@ import ActivityList from './activitylist.js';
  * - `tripId` {string} The trip's ID. This is sent to the component through the URL. 
  */
 class ViewActivities extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state = { addingActivity : false }
+    this.doneAddingActivity = this.doneAddingActivity.bind(this);
```

Comment:
Nit: Why bind only this member function and not the other two? There may be a reason but if not I would make them all the normal member functions and bind them all or all with the experimental syntax and no binding.